src/main/scala/io/movies/api/MovieServiceHttp4s.scala

```diff
@@ -0,0 +1,55 @@
+package io.movies.api
+
+import io.movies.model._
+import cats.effect._
+import cats.syntax.all._
+import io.movies.services.MovieService
+import org.http4s._
+import org.http4s.dsl.Http4sDsl
+import org.http4s.server.blaze._
+import org.http4s.circe._
+import io.circe.generic.auto._
+
+class MovieServiceHttp4s[F[_]: Sync: MovieService]
+  (implicit monadThrow: MonadThrow[F])
+    extends Http4sDsl[F] {
+  val ms: MovieService[F] = implicitly[MovieService[F]]
+  implicit val decoder: EntityDecoder[F, Movie] = jsonOf[F, Movie]
+  implicit val encoder: EntityEncoder[F, RegisteredMovie] = jsonEncoderOf[F, RegisteredMovie]
+  implicit val encoderList: EntityEncoder[F, List[RegisteredMovie]] = jsonEncoderOf[F, List[RegisteredMovie]]
+
+  val MOVIESERVICE = "MovieService"
+  val CREATE = "Create"
+  val READ = "Read"
+  val root = Root / MOVIESERVICE
```

Comment:
I don't think you need the `Create` or `Read` path segments in the endpoint's URLs. The REST resource naming convention points out you should use a resource name and HTTP methods to represent all the allowed operations. For example:
```
GET  /movies - To return a list of movies
GET  /movie/{movieId} - To return the movie identified by `movieId`
POST /movie - To register a new movie
```

Check out this [website](https://restfulapi.net/resource-naming) for further information

Reply:
See https://github.com/gutiory/MoviesServer/pull/1/commits/3d756dcf8d76e19baabd1e292fcba68df7216360